subsys/caf/modules/ble_state.c

```diff
@@ -34,24 +34,26 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_CAF_BLE_STATE_LOG_LEVEL);
 
 
 struct bond_find_data {
-	bt_addr_le_t peer_address;
-	uint8_t peer_id;
-	uint8_t peer_count;
+	const bt_addr_le_t *peer_address;
+	bool peer_bonded;
+	uint8_t bond_cnt;
 };
 
 static struct bt_conn *active_conn[CONFIG_BT_MAX_CONN];
 
 
-static void bond_find(const struct bt_bond_info *info, void *user_data)
+static void bond_check_cb(const struct bt_bond_info *info, void *user_data)
 {
-	struct bond_find_data *bond_find_data = user_data;
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	struct bond_find_data *data = user_data;
 
-	if (bond_find_data->peer_id == bond_find_data->peer_count) {
-		bt_addr_le_copy(&bond_find_data->peer_address, &info->addr);
+	data->bond_cnt++;
+	if (!bt_addr_le_cmp(&info->addr, data->peer_address)) {
+		data->peer_bonded = true;
 	}
 
-	__ASSERT_NO_MSG(bond_find_data->peer_count < UCHAR_MAX);
-	bond_find_data->peer_count++;
+	bt_addr_le_to_str(&info->addr, addr_str, sizeof(addr_str));
```

Comment:
this should be disable in non-debug builds, or when the logger or printk is not enabled

Reply:
Changed. Fixes also other places in `ble_state.c` where similar issues were observed.